src/App.tsx

```diff
@@ -1,24 +1,301 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import * as clientService from './api/Todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { FilterOptions } from './types/FilterOptions';
 
-const USER_ID = 0;
+const USER_ID = 8;
+
+const getFilteredTodos = (todos: Todo[], filterBy: FilterOptions) => {
+  switch (filterBy) {
+    case FilterOptions.Active:
+      return todos.filter(todo => !todo.completed);
+    case FilterOptions.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterOptions.All);
+  const filteredTodos = getFilteredTodos(todos, filterBy);
+  const uncompletedTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const [loadingTodosId, setLoadingTodosId] = useState<number[]>([]);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorMasage, setErrorMasage] = useState(Errors.allGood);
+  const titleField = useRef<HTMLInputElement>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const reset = () => {
+    setTodoTitle('');
+    setErrorMasage(Errors.allGood);
+    setTemporaryTodo(null);
+    setIsDisabled(false);
+    titleField.current?.focus();
+  };
+
+  function loadTodos() {
+    clientService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorMasage(Errors.cantGetArray));
+  }
+
+  const addProcessing = (id: number) => {
+    setLoadingTodosId(current => [...current, id]);
+  };
+
+  const removeProcessing = (idToRemove: number) => {
```

Comment:
I guess you can simplify it - after any action, you should make this array empty. No need to filter out ids one by one